checker/src/main/java/org/checkerframework/checker/signature/qual/FieldDescriptorForPrimitive.java

```diff
@@ -0,0 +1,20 @@
+package org.checkerframework.checker.signature.qual;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Target;
+import org.checkerframework.framework.qual.ImplicitFor;
+import org.checkerframework.framework.qual.SubtypeOf;
+
+/**
+ * Represents a field descriptor (JVM type format) for a primitive as defined in the <a
+ * href="https://docs.oracle.com/javase/specs/jvms/se8/html/jvms-4.html#jvms-4.3.2">Java Virtual
+ * Machine Specification, section 4.3.2</a>.
+ *
+ * <p>Must be one of B,C,D,F,I,J,S,Z
+ *
+ * @checker_framework.manual #signature-checker Signature Checker
+ */
+@SubtypeOf({FieldDescriptorForArray.class, Identifier.class})
```

Comment:
Seeing `FieldDescriptorForArray` is surprising -- a primitive is not an array.  I think the regex for `FieldDescriptorForArray` should not match primitives, and the name `FieldDescriptorForArray` should be changed to `FieldDescriptorForArrayWithoutPackage`.  If there is a need for the concept currently represented by `FieldDescriptorForArray` (which includes non-arrays), then it should be given a different name.